cadence/contracts/bridge/FlowEVMBridgeUtils.cdc

```diff
@@ -727,6 +781,198 @@ contract FlowEVMBridgeUtils {
+        let isEscrowed = self.isOwner(ofNFT: id, owner: self.getBridgeCOAEVMAddress(), evmContractAddress: erc721Address)
+        assert(isEscrowed, message: "Bridge COA does not own NFT after transfer")
+    }
+
+    /// Mints ERC20 tokens to the recipient and confirms that the recipient's balance was updated
+    ///
+    access(account)
+    fun mustMintERC20(to: EVM.EVMAddress, amount: UInt256, erc20Address: EVM.EVMAddress) {
+        let toPreBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        // Mint tokens to the recipient
+        let mintResult: EVM.Result = FlowEVMBridgeUtils.call(
+            signature: "mint(address,uint256)",
+            targetEVMAddress: erc20Address,
+            args: [to, amount],
+            gasLimit: 15000000,
+            value: 0.0
+        )
+        assert(mintResult.status == EVM.Status.successful, message: "Tranfer to bridge recipient failed")
+        // Ensure bridge to recipient was succcessful
+        let toPostBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        assert(
+            toPostBalance == toPreBalance + amount,
+            message: "Transfer to bridge recipient failed"
+        )
+    }
+
+    /// Transfers ERC20 tokens to the recipient and confirms that the recipient's balance was incremented and the escrow
+    /// balance was decremented by the requested amount.
+    ///
+    access(account)
+    fun mustTransferERC20(to: EVM.EVMAddress, amount: UInt256, erc20Address: EVM.EVMAddress) {
+        let bridgeCOAAddress = self.getBridgeCOAEVMAddress()
+
+        let toPreBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        let escrowPreBalance = FlowEVMBridgeUtils.balanceOf(
+            owner: bridgeCOAAddress,
+            evmContractAddress: erc20Address
+        )
+
+        // Mint tokens to the recipient
+        let mintResult: EVM.Result = FlowEVMBridgeUtils.call(
+            signature: "transfer(address,uint256)",
+            targetEVMAddress: erc20Address,
+            args: [to, amount],
+            gasLimit: 15000000,
+            value: 0.0
+        )
+        assert(mintResult.status == EVM.Status.successful, message: "Tranfer to bridge recipient failed")
+
+        // Ensure bridge to recipient was succcessful
+        let toPostBalance = FlowEVMBridgeUtils.balanceOf(owner: to, evmContractAddress: erc20Address)
+        let escrowPostBalance = FlowEVMBridgeUtils.balanceOf(
+            owner: bridgeCOAAddress,
+            evmContractAddress: erc20Address
+        )
+        assert(
+            toPostBalance == toPreBalance + amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
+        )
+        assert(
+            escrowPostBalance == escrowPreBalance - amount,
+            message: "Transfer to bridge COA failed - cannot bridge FT without bridge escrow"
```

Comment:
I feel like these error messages should be a little more descriptive about the issue that is happening, and mention the balances not being correct. What do you think?

Reply:
Just pushed some updates to address